crowbar_framework/app/models/proposal.rb

```diff
@@ -0,0 +1,45 @@
+class Proposal < ActiveRecord::Base
+  # FIXME: add proper i18n to errors
+
+  class TemplateMissing < StandardError; end
+  class TemplateInvalid < StandardError; end
+
+  serialize :properties, JSON
+
+  validates :name, :barclamp, :properties, :presence => true
```

Comment:
I would prefer this formatting:

```
validate :name, :name_not_on_blacklist, presence: true, uniqueness: { scope: :barclamp }
validate :barclamp, :properties, presence: true
```

That way you can see directly all validations for the name attribute.